common/djangoapps/third_party_auth/api/tests/test_views.py

```diff
@@ -130,3 +135,83 @@ def test_list_connected_providers__withapi_key(self, api_key, target_user, expec
         if expect_result == 200:
             self.assertIn("active", response.data)
             self.assertItemsEqual(response.data["active"], self.expected_active(target_user))
+
+    @ddt.data(
+        (VALID_API_KEY, PROVIDER_ID_TESTSHIB, 200, get_mapping_data_by_usernames(LINKED_USERS)),
+        ("i am an invalid key", PROVIDER_ID_TESTSHIB, 403, None),
+        (None, PROVIDER_ID_TESTSHIB, 403, None),
+        (VALID_API_KEY, 'non-existing-id', 404, []),
+    )
+    @ddt.unpack
+    def test_list_all_user_mappings_withapi_key(self, api_key, provider_id, expect_result, expect_data):
+        url = reverse('third_party_auth_user_mapping_api', kwargs={'provider_id': provider_id})
+        response = self.client.get(url, HTTP_X_EDX_API_KEY=api_key)
+        self.assertEqual(response.status_code, expect_result)
+        if expect_result == 200:
+            self._verify_valid_response(response, ['results', 'count', 'num_pages'], expect_data)
+
+    @ddt.data(
+        ({'usernames': [ALICE_USERNAME, STAFF_USERNAME]}, 200,
+         get_mapping_data_by_usernames([ALICE_USERNAME, STAFF_USERNAME])),
+        ({'remote_ids': [ALICE_USERNAME, STAFF_USERNAME]}, 200,
+         get_mapping_data_by_usernames([ALICE_USERNAME, STAFF_USERNAME])),
+        ({'usernames': [ALICE_USERNAME, CARL_USERNAME, STAFF_USERNAME]}, 200,
+         get_mapping_data_by_usernames([ALICE_USERNAME, STAFF_USERNAME])),
+        ({'usernames': [ALICE_USERNAME], 'remote_ids': [STAFF_USERNAME]}, 200,
+         get_mapping_data_by_usernames([ALICE_USERNAME, STAFF_USERNAME])),
+    )
+    @ddt.unpack
+    def test_user_mappings_with_query_params(self, query_params, expect_result, expect_data):
+        base_url = reverse(
+            'third_party_auth_user_mapping_api', kwargs={'provider_id': PROVIDER_ID_TESTSHIB})
+        params = []
+        for attr in ['usernames', 'remote_ids']:
+            if attr in query_params:
+                params.append('{}={}'.format(attr, ','.join(query_params[attr])))
+        url = "{}?{}".format(base_url, '&'.join(params))
+        response = self.client.get(url, HTTP_X_EDX_API_KEY=VALID_API_KEY)
+        self.assertEqual(response.status_code, expect_result)
+        if expect_result == 200:
+            self._verify_valid_response(response, ['results'], expect_data)
```

Comment:
Why are 'count' and 'num_pages' not included here?
Consider creating a helper function for the last 3 lines since it is repeated in several user-mapping related tests.